packages/next/client/image.tsx

```diff
@@ -369,6 +369,15 @@ export default function Image({
     }
   }
 
+  if (layout !== 'fill' && !unsized) {
+    imgStyle = {
+      ...imgStyle,
+      // @ts-ignore TODO: remove ts-ignore when CSSType supports content-visibility
+      contentVisibility: 'auto',
```

Comment:
I'm thinking that we should set `content-visibility: visible` for images with the `priority` attribute. I believe that `auto` should essentially always do what we want, but just semantically I think it would align more with user expectations if `priority` caused the image to render no matter what.

Reply:
That's a good point, will make that change.

Reply:
I think you're right but maybe a user didn't consider mobile when using the `priority` attribute and the image ends up below the fold? Is there harm in leaving this `auto` always, even when the image is above the fold?

Reply:
The documentation does explicit state that `priority` "Should only be used when the image is visible above the fold." Is our philosophy usually "protect the user" from these mistakes, or do we assume that users follow the documentation?

Reply:
If we can protect the user, then we should.

We favor good defaults instead of user configuration.

Reply:
Re: this discussion + syncing offline with @atcastle , I've made the change to only apply this for non lazy loaded images.